build_tools/github_actions/runner/gcp/canary_update_instance_group.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# This is a WIP script for canarying updates to managed instance groups. It
+# isn't really a proper script, since most everything is hardcoded. Mostly it's
+# a way to document the necessary commands. In due course, it may be turned into
+# a proper script, probably in Python, but for now just edit the environment
+# variables at the top.
```

Comment:
A playbook / markdown file with some example scenarios of how to use these scripts would help

Reply:
Yeah, for sure.

Reply:
The "overall shape" of this PR seems good to me, but I haven't really worked through what using this will be like, so that's why I suggested a playbook.

A few questions that could help guide that:
* What should the steady state look like? Is there a page to view what runners / groups are active? Who has access to that page?
* If making changes to the runners / groups, what's the procedure? Push a canary, trigger some builds on it, wait for results, push canary to stable?
* What sorts of things can go wrong, and how would we troubleshoot?

Reply:
How's that?

Reply:
Great! Thanks :)